### DISCUSSION TITLE
Customizing console_title_template to check against variables

### DISCUSSION THREAD

**Discussion by Momoyo** [JanDeDobbeleer/oh-my-posh#5055] [Troubleshoot] [answered]
I'm having some difficulties parsing out the way the JSON if statements work and how to incorporate variables in them. 

Doing something like this
`{
  "myGlobals" : {
     "myHomeRoot" : C:\Users\myUserName\" // this is an example of a directory
   },
  "console_title_template": "{{.Folder}} {{ if .myGlobals.myHomeRoot }} :: ~ {{ end }} :: TESTING",
}`
I know the if conditional, only checks whether the value is valid or exists. But what I'd like is to compare the PWD against my variable and display ~ or an nerdfont icon or whatever to distinguish it. 

Something like this maybe? But I don't quite know if it's even possible with the current Oh-My-Posh API/framework.
`
  "console_title_template": "{{.Folder}} {{ if .PWD == .myGlobals.myHomeRoot }} :: ~ {{ end }} :: TESTING",
`

I also tried doing this through PWSH's scripting language, both through a Prompt function as well as a Register-EngineEvent but nothing quite works the way I'd like it too.
Displaying a variable also doesn't work, both with the dot and the $ sign. Maybe it just doesn't work with console_title_template?

I might be going at this the wrong way.

**Comment by JanDeDobbeleer** [comment, marked as answer]
@Momoyo if I get his right, you're trying to use a global `var` in the configuration like this:


```json
"var" : {
    "myGlobals" : { "myHomeRoot" : "C:\\Users\\myUserName" }
}
```

the try to access it using:

```json
"console_title_template": "{{.Folder}} {{ if .myGlobals.myHomeRoot }} :: ~ {{ end }} :: TESTING"
```

The syntax is incorrect as you still need to tell oh-my-posh to get that context from the `.Var` entry using `{{ .Var.myGlobals.myHomeRoot }}`

And then you can check if the current folder is the same as the one in your variable using:

```json
"console_title_template": "{{ if eq .Var.myGlobals.myHomeRoot .AbsolutePWD }}yes{{ else }}no{{ end }} :: ~ :: TESTING"
```

I noticed the absolute `PWD` value wasn't available just yet, you can expect that with this [PR](https://github.com/JanDeDobbeleer/oh-my-posh/pull/5067).

**Comment by Momoyo** [comment]
Ahhh. Thank you so much! Yes, I was looking for the `eq` operator but it was hard to test the syntax to see how it works. When just accessing the variable gave an empty string it. Turns out that was also the wrong way to access the variable.

And yeah PWD isn't absolute but it would've worked for my case because I just wanted to do fancy little unicode icons in certain directories and use the `.Folder` for others. 

I had a little workaround with my terminal Wezterm, it felt easier to just send the PWD and parse the title there in Lua. But it's much more straightforward this way and the absolutePWD will definitely save me some pesky edgecases! Thank you for the PR and your help. I appreciate it!!
